code/game/objects/items.dm

```diff
@@ -525,19 +567,46 @@ var/list/global/slot_flags_enumeration = list(
 	M.eye_blurry += rand(3,4)
 	return
 
+//As long as was_bloodied is not cleared, the object will still have residual blood traces
 /obj/item/clean_blood()
 	. = ..()
 	if(blood_overlay)
-		overlays.Remove(blood_overlay)
+		for (var/V in overlays)
+			var/image/I = V
+			if (I.name == "blood")
+				overlays.Remove(I)
+
+	if(istype(src, /obj/item/clothing/gloves))

```

Comment:
May be a better idea to refactor this as:
```
/obj/item/clothing/gloves/clean_blood()
    . = ..()
    src.transfer_blood = 0
```

That's what parenting is for.

Then again, source code already uses this method. Sooo. /ASCII-shrug, I guess.